lib/core/utils/preload-cssom.js

```diff
@@ -277,34 +160,355 @@ axe.utils.preloadCssom = function preloadCssom({
 			root,
 			priority
 		};
+	};
+}
+
+/**
+ * Process results from `loadCssom` queues of all root nodes
+ * NOTE:
+ * using `axe.utils.queue` from various `loadCssom` paths, returns a nested array of arrays
+ * -> flatten arrays
+ * -> filter `undefined` results
+ *
+ * @method processCssomAssets
+ * @private
+ * @param {Array<Object>} assets CSSOM assets for each root
+ * @returns {Object} CSSOM object
+ */
+function processCssomAssets(nestedAssets) {
+	const result = [];
+
+	nestedAssets.forEach(item => {
+		if (Array.isArray(item)) {
+			result.push(...processCssomAssets(item));
```

Comment:
Instead of using a single `axe.utils.queue`, which gets passed around the `loadCssom` code path (fetching non cross-origin stylesheets or cross-origin stylesheets), each path now creates a queue of queues, which is why there is a need to recursively flatten the results.

Reply:
Right, but I don't see why this recusion is needed. Can't we just do:

```
nestedAssets.forEach(item => {
  if (Array.isArray(item)) {
    result.push(...item)
  } else {
    result.push(item)
  }
})
```